portal-impl/src/com/liferay/portal/action/UpdateLicenseAction.java

```diff
@@ -41,47 +44,46 @@ public ActionForward execute(
 			HttpServletResponse httpServletResponse)
 		throws Exception {
 
-		// PLACEHOLDER 01
-		// PLACEHOLDER 02
-		// PLACEHOLDER 03
-		// PLACEHOLDER 04
-		// PLACEHOLDER 05
-		// PLACEHOLDER 06
-		// PLACEHOLDER 07
-		// PLACEHOLDER 08
+		if (!_isOmniAdmin(httpServletRequest)) {
+			httpServletResponse.sendRedirect(
+				PortalUtil.getPathContext() + "/c/portal/layout");
 
-		if (_isValidRequest(httpServletRequest)) {
-			String cmd = ParamUtil.getString(httpServletRequest, Constants.CMD);
+			return null;
+		}
+
+		if (Objects.equals(httpServletRequest.getMethod(), "GET")) {
+			return actionMapping.getActionForward("portal.license");
+		}
 
-			String clusterNodeId = ParamUtil.getString(
-				httpServletRequest, "clusterNodeId");
+		if (!_isCSRFTokenValid(httpServletRequest, httpServletResponse)) {
```

Comment:
I think you can merge this with the previous if because I think we should never reach the `return null` part here also it might not be the best to return a `null`.